drivers/td-req.c

```diff
@@ -770,6 +771,8 @@ tapdisk_xenblkif_make_vbd_request(struct td_xenblkif * const blkif,
         tapreq->prot = PROT_READ;
         vreq->op = TD_OP_WRITE;
         break;
+    case BLKIF_OP_FLUSH_DISKCACHE:
```

Comment:
Should we have a sanity check here that number of segments is 0? Also should we have some stats counters so we can tell whether the feature is being used or not?

Reply:
There is, see line 788-791 below.

Reply:
Isn't that test the other way around? if the number of segments is zero check that it is one of the flush operations.
But if you get a flush disk cache with a non-zero segments, what happens then?

Reply:
You can't, it's a dataless operation.

Reply:
This needs to move, a new change is being updated and tested